### DISCUSSION TITLE
Changie tool observations and required developer perspective

### DISCUSSION THREAD

**Discussion by nikhil-swamix** [miniscruff/changie#684] [Ideas] [answered]
#611 reference continued

@miniscruff pardon the ambiguity the idea behind the tool was to offload the cognitive workload of developer to LLM, sometimes user may forget to fully summarize the changes. i was going through https://docs.continue.dev/changelog#intellij-changelog, and few idea striked.
here the tool can compare the <x><y><z> version in package and `git diff` per commit on main branch, and prepend it to `changelog.md` . it is supposed to be used in sync with this tool using bash pipe `smartcommit.py | changie -` similar logic or cli use options from https://changie.dev/cli/changie_new/. the commits serves like additional metadata for better change logs...

Questions:
 
A. can changie edit previous messages or to be manually edit in generated file?
B. how many step required to create a change log, im assuming 3 after watching video.
C. are there any edge cases currently changie does not handle but occured during regular use ? (excluding current thread of text overflow issue)
D. do you identify any drawbacks with my proposed approach, given the development patterns in large+small repos (important).

**Comment by miniscruff** [comment, marked as answer]
A: you can edit the generated yaml file under "changes_dir/unreleased_dir" by default `.changes/unreleased`.
B: The step is usually, repeat `changie new` for how many changes you plan, then `changie batch` to create a new release notes file, you can edit this file to your hearts content such as adding images or videos separate from `changie new`. Then finally `changie merge` to generate the final CHANGELOG.md file. Or whatever file you are using.
C: I know there are issues around dependabot but personally I just skip them entirely. The other main issue I know happens are those who want to reference pull requests rather then issues. But I tend to use issues and not PRs so I don't run into that.
D: I think you are combining the changelogs for ALL changes between releases and a single fragment generated by `changie new` for a single feature/fix. That is, usually you create a changie fragment during your workflow as part of your PR or merge request. Then when it comes to releasing, you already have all your notes for you, no generation needed. When you are working on the feature you have all the context you need to write the release notes at the time, you don't need to go back through the commits to generate anything. Like I said before, it sounds more like you want to work with conventional commits to generate release notes, rather then the pre-written notes from changie.
